internal/storage/memory.go

```diff
@@ -359,3 +377,44 @@ func (m *MemoryConnector) InsertDataForBlocks(data []common.BlockData) error {
 	}
 	return nil
 }
+
+func (m *MemoryConnector) DeleteDataForBlocks(chainId *big.Int, blockNumbers []*big.Int) error {
+	blockNumbersToCheck := getBlockNumbersToCheck(QueryFilter{BlockNumbers: blockNumbers})
+	for _, key := range m.cache.Keys() {
+		shouldDelete := isKeyForBlock(key, fmt.Sprintf("block:%s:", chainId), blockNumbersToCheck) ||
+			isKeyForBlock(key, fmt.Sprintf("log:%s:", chainId), blockNumbersToCheck) ||
+			isKeyForBlock(key, fmt.Sprintf("transaction:%s:", chainId), blockNumbersToCheck) ||
+			isKeyForBlock(key, fmt.Sprintf("trace:%s:", chainId), blockNumbersToCheck)
+		if shouldDelete {
+			m.cache.Remove(key)
+		}
+	}
+	return nil
+}
+
+func (m *MemoryConnector) LookbackBlockHeaders(chainId *big.Int, limit int, lookbackStart *big.Int) ([]common.BlockHeader, error) {
+	blockHeaders := []common.BlockHeader{}
+	for _, key := range m.cache.Keys() {
+		if strings.HasPrefix(key, fmt.Sprintf("block:%s:", chainId.String())) {
```

Comment:
why hasPrefix and not equal?

Reply:
we need items lower than the lookbackStart, so we don't have the block number at that point